### PR TITLE
Repository Setup (Frontend CI, Linting, Testing)

### PR DESCRIPTION
**Pull request by christopherkeim**
## Description

This establishes a `frontend/` directory with linting and testing dependencies and scripts, as well as CI. 

Continuous integration is configured to run as separate actions for `frontend` and the Python source code.

This uses the `vite` setup from `magical-space-scientist` - feel free to trim and modify this however you see fit.

## Testing

Installation, linting, and testing pass.

### TypeScript

- `"@types/node": "^20.10.6"`
- `"@types/eslint": "^8.44.7"`
- `"@typescript-eslint/eslint-plugin": "^6.13.1"`
- `"@typescript-eslint/parser": "^6.13.1"`
- `"@types/jest": "^29.5.10"`
- `"typescript": "^5.2.2"`
- `"vite": "^5.0.8"`
- `"glob": "^10.3.10"`
- `"eslint-config-prettier": "^9.0.0"`
- `"eslint": "^8.54.0"`
- `"concurrently": "^8.2.2"`
- `"jest": "^29.7.0"`
- `"ts-jest": "^29.1.1`

### Python

**Project uses `python3.11`**

- `python = ">=3.11, <3.12"`
- `tensorflow = "2.15.0"`
- `boto3 = "^1.34.22"` AWS SDK

## Scripts added (`npm`)

- `"lint": "eslint src/"`
- `"test": "jest"`
- `"test:watch": "jest --watch"`
- `"test:watch:report": "jest --watch --coverage >/dev/null 2>&1"`
- `"test:verbose": "jest --verbose"`
- `"test:coverage": "jest --coverage"`

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] All new and existing tests are passing.

### CODE REVIEW

File: .github/workflows/python-ci.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ env:
   POETRY_VIRTUALENVS_IN_PROJECT: true
 
 jobs:
-  build:
+  Build:
     runs-on: ubuntu-latest
     container: python:3.10-buster
```
Comment (kensonjohnson) [inline .github/workflows/python-ci.yaml]: Let's try with Python 3.11

**Review by kensonjohnson — CHANGES_REQUESTED** [review]
Looks good. Resolve once you verify Python 3.11 will work.

File: .github/workflows/python-ci.yaml (hunk 1 above)
Comment (christopherkeim) [reply .github/workflows/python-ci.yaml]: Confirmed successfully installations (local and CI) with `python3.11`. Includes `tensorflow` as well.

**Review by kensonjohnson — APPROVED** [review]
:shipit:

**Pull request merged**